File: src/magfi.py
import requests

class Magfi:
	def __init__(
			self,
			country_code: str = "ru") -> None:
		self.api = "http://api.magfiads.com/app"
		self.identity_toolkit_api = "https://www.googleapis.com/identitytoolkit/v3/relyingparty"
		self.headers = {
			"user-agent": "Dart/2.13 (dart:io)"
		}
		self.user_id = None
		self.id_token = None
		self.country_code = country_code
		self.api_key = "AIzaSyAEf31Du8kcON5uudUfSRHNZ6u8Ydg6wZ0"

	def login(
			self,
			email: str,
			password: str) -> dict:
		data = {
			"email": email,
			"password": password,
			"returnSecureToken": True
		}
		response = requests.post(
			f"{self.identity_toolkit_api}/verifyPassword?key={self.api_key}",
			data=data,
			headers=self.headers).json()
		if "idToken" in response:
			self.user_id = response["localId"]
			self.id_token = response["idToken"]
		return response

	def register(
			self,
			email: str,
			password: str) -> dict:
		data = {
			"email": email,
			"password": password
		}
		return requests.post(
			f"{self.identity_toolkit_api}/signupNewUser?key={self.api_key}",
			data=data,
			headers=self.headers).json()

	def get_account_info(self) -> dict:
		data = {
			"idToken": self.id_token
		}
		return requests.post(
			f"{self.api}/getAccountInfo?key={self.api_key}",
			data=data,
			headers=self.headers).json()

	def get_app_version(self) -> dict:
		return requests.get(
			f"{self.api}/app-version",
			headers=self.headers).json()

	def get_main_page(self) -> dict:
		return requests.get(
			f"{self.api}app/main-page?userID={self.user_id}",
			headers=self.headers).json()

	def get_cat_names(self) -> list:
		return requests.get(
			f"{self.api}/app-cat-names",
			headers=self.headers).json()

	def get_user_follow_count(
			self,
			user_id: str) -> dict:
		return requests.get(
			f"{self.api}/follow-count?userID={user_id}",
			headers=self.headers).json()

	def get_user_followers(
			self,
			user_id: str) -> list:
		return requests.get(
			f"{self.api}/followers?userID={user_id}",
			headers=self.headers).json()

	def get_user_followings(
			self,
			user_id: str) -> list:
		return requests.get(
			f"{self.api}/following?userID={user_id}",
			headers=self.headers).json()

	def get_categories(self) -> list:
		return requests.get(
			f"{self.api}/category-list?country_code={self.country_code}",
			headers=self.headers).json()

	def get_suggested_users(self) -> list:
		return requests.get(
			f"{self.api}/suggested-users?userID={self.user_id}",
			headers=self.headers).json()

	def get_feed(self, page: int = 1) -> list:
		return requests.get(
			f"{self.api}/FeedInfinty?page={page}&userID={self.user_id}",
			headers=self.headers).json()

	def get_trends(self) -> list:
		return requests.get(
			f"{self.api}/app/trends",
			headers=self.headers).json()

	def get_language_region(self) -> list:
		return requests.get(
			f"{Self.api}/language-region?country_code={self.country_code}",
			headers=self.headers).json()

	def change_password(
			self,
			password: str) -> dict:
		data = {
			"returnSecureToken": True,
			"idToken": self.id_token,
			"password": password
		}
		return requests.post(
			f"{self.identity_toolkit_api}/setAccountInfo?key={self.api_key}",
			data=data,
			headers=self.headers).json()

File: src/magfi.js
class Magfi {
	constructor(countryCode = "ru") {
		this.api = "http://api.magfiads.com/app"
		this.identityToolkitApi = "https://www.googleapis.com/identitytoolkit/v3/relyingparty"
		this.headers = {
			"user-agent": "Dart/2.13 (dart:io)"
		}
		this.countryCode = countryCode
		this.apiKey = "AIzaSyAEf31Du8kcON5uudUfSRHNZ6u8Ydg6wZ0"
	}


	async login(email, password) {
		const response = await fetch(
			`${this.identityToolkitApi}/verifyPassword?key=${this.apiKey}`, {
				method: "POST",
				body: JSON.stringify({
					email: email,
					password: password,
					returnSecureToken: true
				}),
				headers: this.headers
			})
		const data = await response.json()
		if ("idToken" in data) {
			this.userId = data.localId
			this.idToken = data.idToken
		}
		return data
	}

	async register(email, password) {
		const response = await fetch(
			`${this.identityToolkitApi}/signupNewUser?key=${this.apiKey}`, {
				method: "POST",
				body: JSON.stringify({
					email: email,
					password: password
				}),
				headers: this.headers
			})
		return response.json()
	}

	async getAccountInfo() {
		const response = await fetch(
			`${this.api}/getAccountInfo?key=${this.apiKey}`, {
				method: "POST",
				body: JSON.stringify({
					idToken: this.idToken
				}),
				headers: this.headers
			})
		return response.json()
	}

	async getAppVersion() {
		const response = await fetch(
			`${this.api}/app-version`, {
				method: "GET",
				headers: this.headers
			})
		return response.json()
	}

	async getMainPage() {
		const response = await fetch(
			`${this.api}/app/main-page?userID=${this.userId}`, {
				method: "GET",
				headers: this.headers
			})
		return response.json()
	}

	async getCatNames() {
		const response = await fetch(
			`${this.api}/app-cat-names`, {
				method: "GET",
				headers: this.headers
			})
		return response.json()
	}

	async  getUserFollowers(userId) {
		const response = await fetch(
			`${this.api}/followers?userID=${userId}`, {
				method: "GET",
				headers: this.headers
			})
		return response.json()
	}

	async  getUserFollowings(userId) {
		const response = await fetch(
			`${this.api}/following?userID=${userId}`, {
				method: "GET",
				headers: this.headers
			})
		return response.json()
	}

	async getSuggestedUsers() {
		const response = await fetch(
			`${this.api}/suggested-users?userID=${this.userId}`, {
				method: "GET",
				headers: this.headers
			})
		return response.json()
	}

	async getFeed(page = 1) {
		const response = await fetch(
			`${this.api}/FeedInfinty?page=${page}&userID=${this.userId}`, {
				method: "GET",
				headers: this.headers
			})
		return response.json()
	}

	async getTrends() {
		const response = await fetch(
			`${this.api}/app/trends`, {
				method: "GET",
				headers: this.headers
			})
		return response.json()
	}
}

module.exports = {Magfi}
